api/buckets/service.go

```diff
@@ -1606,33 +1607,87 @@ func (s *Service) Archive(ctx context.Context, req *pb.ArchiveRequest) (*pb.Arch
 		return nil, fmt.Errorf("parsing cid path: %s", err)
 	}
 
-	ffsi, err := s.Collections.FFSInstances.Get(ctx, req.GetKey())
-	if err != nil {
-		return nil, fmt.Errorf("getting ffs instance data: %s", err)
+	var ffsInfo *mdb.FFSInfo
+	var owner crypto.PubKey
+	var isAccount bool
+	if acct := accountFromContext(ctx); acct != nil {
+		ffsInfo = acct.FFSInfo
+		owner = acct.Key
+		isAccount = true
+	} else if user := userFromContext(ctx); user != nil {
+		ffsInfo = user.FFSInfo
+		owner = user.Key
+		isAccount = false
+	}
+
+	createNewFFS := func() error {
+		id, token, err := s.PGClient.FFS.Create(ctx)
+		if err != nil {
+			return fmt.Errorf("creating new ffs instance: %v", err)
+		}
+		if isAccount {
+			_, err = s.Collections.Accounts.UpdateFFSInfo(ctx, owner, &mdb.FFSInfo{ID: id, Token: token})
+		} else {
+			_, err = s.Collections.Users.UpdateFFSInfo(ctx, owner, &mdb.FFSInfo{ID: id, Token: token})
+		}
+		if err != nil {
+			return fmt.Errorf("updating user/account with new ffs information: %v", err)
+		}
+		return nil
+	}
+
+	tryAgain := fmt.Errorf("new ffs instance created, please try again in 30 seconds to allow time for wallet funding")
+
+	// case where account/user was created before bucket archives were enabled.
+	// create a ffs instance for them.
+	if ffsInfo == nil {
+		if err := createNewFFS(); err != nil {
+			return nil, err
+		}
+		return nil, tryAgain
 	}
 
-	ctxFFS := context.WithValue(ctx, powc.AuthKey, ffsi.FFSToken)
+	ctxFFS := context.WithValue(ctx, powc.AuthKey, ffsInfo.Token)
+
+	defConf, err := s.PGClient.FFS.DefaultStorageConfig(ctxFFS)
+	if err != nil {
+		if err.Error() != "auth token not found" {
```

Comment:
Future idea: Maybe create an `ErrTokenNotFound` so at least can be stringified using the client library of powergate? At least if we change the msg **and** client is kept up to date with powd, we shouldn't need to remember changing this string. 

(General idea though, not really specific to this case)

Reply:
Yea I was thinking we need similar improvements in general. Been looking at gRPC's support for custom error metadata. Handy package [here](https://godoc.org/google.golang.org/genproto/googleapis/rpc/errdetails)